Add cpp examples (#435)
nmslib/hnswlib: README.md

        
  
#### Python bindings examples
[See more examples here](examples/python/EXAMPLES.md):
* creating index, inserting elements, searching, serialization/deserialization

> * filtering during the search

Reviewer: ```suggestion
* filtering during the search with a boolean function 
```

> * delete elements and reusing the memory of the deleted elements when new elements are being added

Reviewer: ```suggestion
* deleting the elements and reusing the memory of the deleted elements for newly added elements
```

        
  
#### Python bindings examples

> [See more examples here](examples/python/EXAMPLES.md)

Reviewer: Maybe give a one-two word description of the examples

Author: done

print("Recall for two batches:", np.mean(labels.reshape(-1) == np.arange(len(data))), "\n")
```

#### C++ examples
[See examples here](examples/cpp/EXAMPLES.md):
* creating index, inserting elements, searching, serialization/deserialization

> * filtering during the search

Reviewer: ```suggestion
* filtering during the search with a boolean function
```

> * delete elements and reusing the memory of the deleted elements when new elements are being added

Reviewer: ```suggestion
* deleting the elements and reusing the memory of the deleted elements for newly added elements
```
